lib/schedulers/serialDispatcher.js

```diff
@@ -11,13 +11,12 @@ class SerialDispatcher {
 
     _setupQueue(queue) {
 
-        const self = this;
         const context = {
             queue,
             buffer: [],
             intervalRef: setIntervalAsync(async () => {
 
-                const ctx = self._queues.get(queue);
+                const ctx = this._queues.get(queue);
```

Comment:
I'm not positive this will work since `this` is not the same as the one in the outer scope.  `setIntervalAsync` runs in a different context when it executes the anon func.